### PR TITLE
=clu #19622 Use full address in ClusterShardingStats

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -555,7 +555,7 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
         ShardRegion.ClusterShardingStats(allRegionStats.map {
           case (region, stats) ⇒
             val address: Address =
-              if (region == self) Cluster(context.system).selfAddress
+              if (address.hasLocalScope && address.system == cluster.selfAddress.system) cluster.selfAddress
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: address refers to the address you are assigning here, so this does not compile

**Comment by johanandren** [comment]
Refs #19622

**Comment by talp-traiana** [comment]
Ohhh, riiight. I compiled akka-cluster instead of akka-cluster-sharding... Sorry.

**Comment by johanandren** [comment]
Would be a good idea (and pretty easy) to cover this in the tests also, in `ClusterShardingGetStatsSpec` you should be able to just add assertions on the address in the stats messages.

Note that those are multi-jvm tests and you would run it with:
`akka-cluster-sharding/multi-jvm:testOnly akka.cluster.sharding.ClusterShardingGetStatsSpec`

**Comment by talp-traiana** [comment]
@johanandren OK, how's this look?

**Comment by johanandren** [comment]
LGTM!

**Comment by jamesmulcahy** [comment]
LGTM. I've tested this and it fixes the problem I originally encountered.  Thanks @talp-traiana

**Pull request merged**